### DISCUSSION TITLE
Moving lines cause custom line controls to reset position of points.

### DISCUSSION THREAD

**Discussion by rockerBOO** [fabricjs/fabric.js#7917] [Q&A] [answered]
I made some custom controls for lines using the example I found for polygons. The controls work for the 2 points fine. When moving the line using the `dragHandler` it causes the points for the lines to incorrectly reset to a previous position. This also happens when loading points from stored positions in my application, but keeping this to what I am showcasing below. 

I'd expect the position of the points to not change after moving the line using the default controls. 

https://codepen.io/rockerboo/pen/KKQKePo

https://user-images.githubusercontent.com/15027/166160364-46b0d7d0-74b1-4ecd-a523-c95d70b2ce42.mp4

Any ideas on what I'm missing? If I lock the movement (`lockMovementX`, `lockMovementY`) to prevent the drag handler from moving I don't have these issues. Willing to release these controls to the community.

**Comment by ShaMan123** [comment, marked as answer]
This is a known issue. I haven't found what causes it yet.
It seems related to the polygon issues with stroke.

**Reply by rockerBOO** [reply]
I have gotten polygons to work perfectly, not having this issue. Maybe this can help to identify polygon issues?

Example: https://codepen.io/rockerboo/pen/vYdYayR

**Reply by ShaMan123** [reply]
I suspect it's `Line#_setWidthHeight` but I'm guessing.
If I were you I would stay away from Line and use Path instead.

**Reply by ShaMan123** [reply]
> I have gotten polygons to work perfectly, not having this issue. Maybe this can help to identify polygon issues?
> 
> Example: https://codepen.io/rockerboo/pen/vYdYayR

Maybe PR to the website?
https://github.com/fabricjs/fabricjs.com
https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#working-on-fabricjscom
https://github.com/fabricjs/fabricjs.com/blob/gh-pages/posts/demos/_posts/2020-01-15-custom-controls-polygon.md

**Reply by ShaMan123** [reply]
Once you set `strokeUniform: true` and scale the polygons the issue surfaces.

**Reply by ShaMan123** [reply]
> If I were you I would stay away from Line and use Path instead.

Even better: use Polygon with 2 points

![ezgif com-gif-maker (5)](https://user-images.githubusercontent.com/34343793/166164952-40b259bd-1f94-481e-9e6a-7712fa32a6ae.gif)

**Comment by ShaMan123** [comment]
I have fixed the `strokeUniform` issue thanks to #7639
